src/components/DataTypes/index.ts

```diff
@@ -25,7 +25,7 @@ export const getDataModel = (schema?: JSONSchema) => {
 		return arrayType;
 	}
 	if (type === 'string') {
-		if (format === 'date-time') {
+		if (format?.endsWith('-time')) {
```

Comment:
Changed check to be able to also support custom date-time format (e.g. custom extraFormat)

Reply:
@JSReds this is too hacky for my taste, let's just create an array of supported widgets for now, it is a bit cleaner in the short-term

Reply:
Can format sometimes just be `'time'`?

Reply:
@sradevski we cannot know the supported widgets since we can add custom widgets in autoUI context. (e.g. in the device list `last_connectivity_event` needs a custom `device-elapsed-time` widget, and it also needs to display the calendar component)

@grahammcculloch sure it makes sense, I was skipping it because I thought we could implement a different filter for time. If it's just time, I don't expect a calendar in the filters but maybe just the time selection. Does this make sense ?

Reply:
@JSReds yep - that makes sense.

Reply:
@JSReds I think there is a limited number of ways how we can represent time (and it should be limited). I don't think we need to have device elapsed time, what does that even mean? Is there some different type of time I don't know about? 😄

Reply:
@sradevski maybe in the future we will have a new type of time 😄 , but it's not what I was considering in this context. There are no differences in time but in presentation. In a normal `elapsed-time` representation you would see something like `14 days ago` in a device elapsed time custom widget you would see `Online (for 4 months) `

Reply:
The solution we came up with by talking about it during a call is to rename the widget `elapsed-date-time` so we can check if the format `endsWith('date-time')` which seems more appropriate.